docs/development/design/fsm.md

```diff
@@ -0,0 +1,57 @@
+# Nexd State Machine
+
+> [Issue #885](https://github.com/nexodus-io/nexodus/issues/885)
+
+## Summary
+
+State management in nexd is becoming more complicated.
+We need a mechanism to allow us to simplify that logic and not rely on crash/restart to retrigger things.
+
+## Proposal
+
+I propose that we implement a Finite State Machine inside of nexd that operates as follows:
+
+```mermaid
+---
+title: Nexd State Machine
+---
+stateDiagram-v2
+    [*] --> IDLE
+    IDLE --> AUTH: "login"
+    AUTH --> READY: "login_success"
+    AUTH --> IDLE: "login_failed"
+    READY --> PEERING: "start"
+    READY --> IDLE: "logout"
+    PEERING --> READY: "stop"
+    PEERING --> READY: "peering_error"
+    PEERING --> IDLE: "authentication_error"
+    PEERING --> IDLE: "logout"
+```
+
+nexd may be in one of four states:
+
+1. IDLE: No configuration has been provided
+2. AUTH: Performing authentication, either interactively or via token
+3. READY: Authenticated and ready to peer
+4. PEERING: Actively Peering
+
+To provide information to the user about the current state, a `nexctl nexd status` command will be added. This can inform the user of the current state and what to do next.
```

Comment:
```suggestion
To provide information to the user about the current state, a `nexctl nexd status` command will be updated to include the new states. This can inform the user of the current state and what to do next.
```